tests/testNonlinearFactor.cpp

```diff
@@ -101,6 +101,82 @@ TEST( NonlinearFactor, NonlinearFactor )
   DOUBLES_EQUAL(expected,actual,0.00000001);
 }
 
+/* ************************************************************************* */
+TEST(NonlinearFactor, Weight) {
```

Comment:
@dellaert I added a unit test for the `weight` function. Can you please check it out?

Reply:
@yetongumich and @ProfFan 
You guys did the deep dive into robust models. Can you please weigh in?

Reply:
I see no problem, @yetongumich what do you think?

Reply:
LGTM